frontend/store/network.ts

```diff
@@ -114,8 +114,15 @@ export const actions = {
         await dispatch('surplus/setup', undefined, { root: true });
         await dispatch('debt/setup', undefined, { root: true });
     },
-    async setupNetworks({ rootGetters, commit }: ActionContext<State, State>) {
-        const rpcUrl = rootGetters['preferences/getRpcUrl'] || process.env.RPC_URL;
+    async setupNetworks({ rootGetters, commit, dispatch }: ActionContext<State, State>) {
+        let rpcUrl: string | undefined;
+        if (process.env.RPC_URL && !rootGetters['preferences/getWasEnvRpcUrlUsed']) {
```

Comment:
Why do you need to introduce and check `wasEnvRpcUrlUsed`?  If `RPC_URL` is provided, it should overwrite locally stored value in any case

Reply:
Also, please don't forget to tick [the requests](https://github.com/sidestream-tech/unified-auctions-ui/pull/573#issuecomment-1410027409) that were implemented